cdap-examples/Purchase/src/main/java/co/cask/cdap/examples/purchase/PurchaseHistoryBuilder.java

```diff
@@ -134,8 +134,9 @@ public void reduce(Text customer, Iterable<Text> values, Context context)
 
       PurchaseHistory purchases = new PurchaseHistory(customer.toString(), userProfile);
       int numPurchases = 0;
-      for (Text val : values) {
-        purchases.add(new Gson().fromJson(val.toString(), Purchase.class));
+      for (Purchase val : values) {
+        purchases.add(new Purchase(val.getCustomer(), val.getProduct(), val.getQuantity(), val.getPrice(),
```

Comment:
Is this needed? Or can we just use ``val`` directly? Or write ``purchases`` directly to the context? Is it necessary to do defensive copy here? If so, can you add a constructor Purchase(Purchase other)? I think that is nicer.